templates/default/client.rb.erb

```diff
@@ -47,20 +47,20 @@ ohai.plugin_path << "<%= node["ohai"]["plugin_path"] %>"
 ohai.disabled_plugins = [<%= @ohai_disabled_plugins.map { |k| k.match(/:/) ? k.gsub(/^:/, '').to_sym.inspect : k.inspect }.join(",") %>]
 <% end -%>
 
-<% if !@start_handlers.empty? || !@report_handlers.empty? || !@exception_handlers.empty? -%>
+<% if !@start_handlers.nil? || !@report_handlers.nil? || !@exception_handlers.nil? -%>
```

Comment:
most likely all that should happen here is that the original `empty?` check here should be deleted along with its associated `end`.

the way this will work after the change is that 2 out of 3 of these could be nil, but as long as one isn't, then it'll enter through this if statement and then later those nil values will blow up in their `#each` statement.

it also isn't necessary at all to protect with `empty?` checks like this since `each` just does nothing when passed an empty array.

you could change the conditional so that if any of the arrays were `empty?` that it would not fire (`unless @start_handlers.nil? || @report_handlers.nil? || @exception_handlers.nil?`) but then if you have one nil value the other non-nil values do not get rendered which is another kind of bug.

so its best to do the `nil?` checks down below around the loops, and then let `#each` deal with the `empty?` case implicitly.